plugin/acl/setup.go

```diff
@@ -0,0 +1,315 @@
+package acl
+
+import (
+	"bufio"
+	"fmt"
+	"net"
+	"os"
+	"strings"
+	"unicode"
+
+	"github.com/caddyserver/caddy"
+	"github.com/coredns/coredns/core/dnsserver"
+	"github.com/coredns/coredns/plugin"
+	"github.com/coredns/coredns/plugin/acl/filter"
+	"github.com/coredns/coredns/plugin/metrics"
+	"github.com/coredns/coredns/plugin/pkg/log"
+	"github.com/miekg/dns"
+)
+
+const (
+	// QtypeAll is used to match any kinds of DNS records type.
+	// NOTE: The value of QtypeAll should be different with other QTYPEs defined in miekg/dns.
+	QtypeAll uint16 = dns.TypeANY
+)
+
+var (
+	// PrivateNets defines all ip addresses reserved for private networks.
+	// i.e., 10.0.0.0/8, 172.16.0.0/12, 192.168.0.0/16.
+	PrivateNets = []string{"10.0.0.0/8", "172.16.0.0/12", "192.168.0.0/16"}
+)
+
+func init() {
+	caddy.RegisterPlugin("acl", caddy.Plugin{
+		ServerType: "dns",
+		Action:     setup,
+	})
+}
+
+func setup(c *caddy.Controller) error {
+	a, err := parseACL(c)
+	if err != nil {
+		return err
+	}
+
+	dnsserver.GetConfig(c).AddPlugin(func(next plugin.Handler) plugin.Handler {
+		a.Next = next
+		return a
+	})
+
+	// Register all metrics.
+	c.OnStartup(func() error {
```

Comment:
this will panic when done multiple times, are there tests for that?

Reply:
> this will panic when done multiple times, are there tests for that?

Could you please provide me some context about panic., cuz the metrics work well in my machine.

![image](https://user-images.githubusercontent.com/14845825/63530622-68965080-c539-11e9-91ee-c3b4562fc2e4.png)